ee/server/service/software_installers.go

```diff
@@ -273,9 +273,11 @@ func (svc *Service) GetSoftwareInstallResults(ctx context.Context, resultUUID st
 		return nil, err
 	}
 
-	// Team specific auth check
-	if err := svc.authz.Authorize(ctx, &fleet.HostSoftwareInstallerResultAuthz{HostTeamID: res.HostTeamID}, fleet.ActionRead); err != nil {
-		return nil, err
+	if res.HostDeletedAt == nil {
```

Comment:
Note that if the host was deleted, the only authorization done is "list hosts" above. Should there be a different authz for that case?

Reply:
I tried to come up with something else but I couldn't, the other alternative is to keep some host information around.

We never merged this, and if you remember: but at some point we tried to solve a similar problem, (but for users that were deleted), we were considering having a table like `deleted_user_data` that kept information like this.

Something like that would be more tricky for hosts because the table is so huge that there will be the temptation to slowly move all the fields.